awscli/customizations/s3/subcommands.py

```diff
@@ -475,6 +475,8 @@ def _run_main(self, parsed_args, parsed_globals):
             path = path[5:]
         bucket, key = find_bucket_key(path)
         if not bucket:
+            if key:
+                return 0
```

Comment:
Probably it'd better to raise a ParamValidation exception here instead of just successfully exit without any output. What do you think about checking both the key and the bucket instead?
```
if not bucket and not key:
    self._list_all_buckets()
```

Reply:
Is it expected?

```
aws s3 ls s3:///random_nonsense
```

Result:

```
Parameter validation failed:
Invalid bucket name "": Bucket name must match the regex "^[a-zA-Z0-9.\-_]{1,255}$" or be an ARN matching the regex "^arn:(aws).*:s3:[a-z\-0-9]+:[0-9]{12}:accesspoint[/:][a-zA-Z0-9\-]{1,63}$"
```

Case: no bucket & no key

```
aws s3 ls s3:///
```

Result:

```
2021-03-31 17:16:51 bucket-1
2019-12-27 13:48:23 bucket-2
2020-09-04 13:36:46 bucket-3
```

Reply:
Yes, I think this the best what we can do. In the first case it's unclear what customer actually needs and we'd better fail than try to guess, in the second one since input is technically empty we may show everything.

Reply:
Got it. I'll update it tonight. Thanks a lot.